Source/WebKit/WebProcess/Extensions/WebExtensionContextProxy.cpp

```diff
@@ -189,15 +189,21 @@ void WebExtensionContextProxy::enumerateFramesAndNamespaceObjects(const Function
 
         auto context = page->isServiceWorkerPage() ? frame->jsContextForServiceWorkerWorld(world) : frame->jsContextForWorld(world);
         auto globalObject = JSContextGetGlobalObject(context);
+
         auto namespaceObject = JSObjectGetProperty(context, globalObject, toJSString("browser").get(), nullptr);
+        if (namespaceObject && JSValueIsObject(context, namespaceObject)) {
+            if (RefPtr namespaceObjectImpl = toWebExtensionAPINamespace(context, namespaceObject)) {
+                function(frame, *namespaceObjectImpl);
+                continue;
```

Comment:
This is wrong, when it finds it, it should use it. This will skip the whole frame.

You want something like this:

```c++
RefPtr namespaceObjectImpl;

auto browserNamespaceObject = JSObjectGetProperty(context, globalObject, toJSString("browser").get(), nullptr);
if (browserNamespaceObject && JSValueIsObject(context, browserNamespaceObject))
    namespaceObjectImpl = toWebExtensionAPINamespace(context, browserNamespaceObject);

if (!namespaceObjectImpl) {
    auto chromeNamespaceObject = JSObjectGetProperty(context, globalObject, toJSString("chrome").get(), nullptr);
    if (chromeNamespaceObject && JSValueIsObject(context, chromeNamespaceObject))
        namespaceObjectImpl = toWebExtensionAPINamespace(context, chromeNamespaceObject);
}

if (!namespaceObjectImpl)
    continue;
```

Reply:
Line 196 calls the function, so it does get used, but I can structure it like this if you prefer.

Reply:
Yep missed that!